### PR TITLE
Implement question hints

### PR DESCRIPTION
**Pull request by gboutry**
Implement the ability to define description as hints for the questions. The goal is to help user from the CLI understand what the question is looking for.

Toggleable with `--hints/--no-hints`.

### CODE REVIEW

File: sunbeam-python/sunbeam/provider/local/steps.py [hunk 1]
```diff
@@ -132,21 +139,29 @@ def prompt_for_nic(self, console: Console | None = None) -> str | None:
...
                 if not agree_nic_up:
                     continue
             if nic_state["up"] and not nic_state["connected"]:
                 agree_nic_no_link = sunbeam.core.questions.ConfirmQuestion(
                     f"WARNING: Interface {nic} is not connected. Are "
-                    "you sure you want to continue?"
+                    "you sure you want to continue?",
+                    description=(
+                        "Interface is not detected as connected to any network. This"
+                        " means it will most likely not work as expected."
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/provider/local/steps.py:152]: Do you think we can inform what does not work as expected? Like external connectivity?

File: sunbeam-python/sunbeam/steps/clusterd.py [hunk 2]
```diff
@@ -60,8 +60,14 @@
 def bootstrap_questions():
     return {
         "management_cidr": questions.PromptQuestion(
-            "Management network (CIDRs, separated by comma)",
+            "Management network",
             default_value=utils.get_local_cidr_by_default_route(),
+            description=(
+                "Management network should be available on every nodes of"
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/steps/clusterd.py:66]: nit: %s/nodes/node in L#66

File: sunbeam-python/sunbeam/utils.py [hunk 3]
```diff
@@ -374,6 +374,17 @@ def decorator(func: click.decorators.FC) -> click.decorators.FC:
     return decorator
 
 
+def click_option_show_hints(func: click.decorators.FC) -> click.decorators.FC:
+    """Common decorator to show hints for questions."""
+    return click.option(
+        "--hints/--no-hints",
+        "show_hints",
+        is_flag=True,
+        default=True,
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/utils.py:383]: Isn't it better for default False and user can set --hints if more info is required?

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]
Minor nits inline

File: sunbeam-python/sunbeam/utils.py (hunk 3 above)
Comment (gboutry) [reply sunbeam-python/sunbeam/utils.py:383]: We had this discussion, and I believe the interactive installation is tailored towards first time users. Veterans will most likely use the manifest and not use the interactive CLI

File: sunbeam-python/sunbeam/provider/local/steps.py (hunk 1 above)
Comment (gboutry) [reply sunbeam-python/sunbeam/provider/local/steps.py:152]: Not sure what you intend.

`Interface not connected to a network means there will be no connectivity` (something like that), seems redundant to me? But maybe needs to be clarified to an user

**Pull request merged**